client/src/redux/reducers/viewSlice.js

```diff
@@ -13,6 +13,18 @@ const _setActiveTripId = (state, action) => {
   _setAppView(state, { payload: AppView.TRIP_OVERVIEW })
 }
 
+const _toggleSidebar = (state) => {
+  state.isSidebarOpen = !state.isSidebarOpen
+}
+
+const _openSidebar = (state) => {
+  state.isSidebarOpen = true
```

Comment:
is it okay to be directly assigning state or should we make immutable updates?

cross checking with: https://redux.js.org/tutorials/fundamentals/part-3-state-actions-reducers#rules-of-reducers

Reply:
Doc suggests approaching as below, by creating copy and updating its value: 

<img width="783" alt="Screenshot 2023-05-27 at 10 36 42 PM" src="https://github.com/cpsc455-bugstorm/TravelersTea/assets/59987667/6afdd8a1-5bf5-4e49-b160-d8c12c71ec77">

Reply:
that is traditional redux. we are using redux toolkit. https://redux-toolkit.js.org/usage/immer-reducers
`Redux Toolkit's [createReducer](https://redux-toolkit.js.org/api/createReducer) and [createSlice](https://redux-toolkit.js.org/api/createSlice) automatically use [Immer](https://immerjs.github.io/immer/) internally to let you write simpler immutable update logic using "mutating" syntax. This helps simplify most reducer implementations.`

Reply:
it allows you to write code that appears to mutate the state directly but it is actually applying the updates without mutating

Reply:
Okay, that's awesome!